lang/cs/Org.Apache.REEF.Driver/Bridge/Events/DriverRestarted.cs

```diff
@@ -17,21 +17,30 @@
  * under the License.
  */
 using System;
+using System.Collections.Generic;
+using Org.Apache.REEF.Driver.Bridge.Clr2java;
 
 namespace Org.Apache.REEF.Driver.Bridge.Events
 {
     internal sealed class DriverRestarted : IDriverRestarted
     {
         private readonly DateTime _startTime;
+        private readonly ISet<string> _expectedEvaluatorIds; 
 
-        internal DriverRestarted(DateTime startTime)
+        internal DriverRestarted(IDriverRestartedClr2Java driverRestartedClr2Java)
         {
-            _startTime = startTime;
+            _startTime = driverRestartedClr2Java.GetStartTime();
+            _expectedEvaluatorIds = new HashSet<string>(driverRestartedClr2Java.GetExpectedEvaluatorIds());
```

Comment:
Oooh, I see. In that case, can we create a copy-on-access? Otherwise, this is not safe, right? Buggy application code can grab the evaluator Ids and change it?